docs/book/src/quinn/data-transfer.md

```diff
@@ -0,0 +1,165 @@
+An instance of this type, together with existing streams, can be found in the [connection][connection] field of [NewConnection].
+
+*Iterate over various streams*
+
+```rust
+async fn iterate_streams(mut connection: NewConnection) -> anyhow::Result<()> {
+    // Iterate unidirectional streams with only the receiving side.
+    while let Some(Ok(recv)) = connection.uni_streams.next().await { }
+    // Iterate bidirectional streams with both sent and receiving side.
+    while let Some(Ok((sent, recv))) = connection.bi_streams.next().await { }
+    // Iterate arrived datagrams.
+    while let Some(Ok(bytes)) = connection.datagrams.next().await { }
+
+    Ok(())
+}
+```
+
+*Open various kinds streams*
+
+```rust
+async fn open_streams(mut connection: Connection) -> anyhow::Result<()> {
+    // Open unidirectional stream.
+    let mut send = connection.
+        open_uni()
+        .await?;
+
+    // Open bidirectional stream.
+    let (send, recv) = connection.
+        open_bi()
+        .await?;
+
+    Ok(())
+}
+```
+
+## Bidirectional Streams
+
+With bidirectional streams data can be sent in both directions, for example, from the connection initiator to the peer and the other way around.
+ 
+*open bidirectional stream*
+
+```rust
+async fn open_bidirectional_stream(connection: Connection) -> anyhow::Result<()> {
+    let (mut send, recv) = connection.
+        open_bi()
+        .await?;
+
+    send.write_all(b"test").await?;
+    send.finish().await?;
+    
+    let received = recv.read_to_end(10).await?;
+
+    Ok(())
+}
+```
+
+*iterate bidirectional stream(s)*
+
+```rust
+async fn receive_bidirectional_stream(mut connection: NewConnection) -> anyhow::Result<()> {
+    while let Some(Ok((sent, recv))) = connection.bi_streams.next().await {
+        // Because it is a bidirectional stream, we can both sent and recieve.
+        println!("request: {:?}", recv.read_to_end(50).await?);
+
+        send.write_all(b"response").await?;
+        send.finish().await?;
+    }
+
+    Ok(())
+}
+```
+
+## Unidirectional Streams 
+
+With unidirectional streams, you can carry data only in one direction, for example, from the initiator of the stream to its peer.
+    
+*open unidirectional stream*
+
+```rust
+async fn open_unidirectional_stream(connection: Connection)-> anyhow::Result<()> {
+    let mut send = connection.
+        open_uni()
+        .await?;
+
+    send.write_all(b"test").await.unwrap();
+    send.finish().await?;
+
+    Ok(())
+}
+```
+
+*iterating unidirectional stream(s)*
+
+```rust
+async fn receive_unidirectional_stream(mut connection: NewConnection) -> anyhow::Result<()> {
+    while let Some(Ok(recv)) = connection.uni_streams.next().await {
+        // Because it is a unidirectional stream, we can only receive not sent back.
+        println!("{:?}", recv.read_to_end(50).await?);
+    }
+
+    Ok(())
+}
+```
+
+## Unreliable Messaging
+
+With unreliable messaging you can transfer data unreliable UDP.
```

Comment:
This isn't really a sentence. Also maybe clarify why unreliability is desirable for some kinds of data, and why datagrams can be a better solution rather than having lots of short unidirectional streams.

Reply:
What is the benefit for many unidirectional streams vs unreliable messaging?

Reply:
Many unidirectional streams get you reliability without ordering (so no head-of-line blocking).

Reply:
Since this is still open, I'm waiting for you to follow up, but perhaps you are waiting for my review? Just realized that it's the end of January and we haven't completely merged your work yet, sorry about that!

Reply:
This is an old comment and some things have changed already.  Tho I will introduce 'Many unidirectional streams get you reliability without ordering (so no head-of-line blocking)' into this text.